src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,133 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int MIN_AGE = 18;
+    private static final int DEFAULT_AGE = 20;
+    private static final String DEFAULT_LOGIN = "admin";
+    private static final String DEFAULT_PASSWORD = "12345678";
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setAge(DEFAULT_AGE);
+        user.setLogin(DEFAULT_LOGIN);
+        user.setPassword(DEFAULT_PASSWORD);
+    }
+
+    @Test
+    void validUser_Ok() {
+        assertEquals(user, registrationService.register(user));
+    }
+
+    @Test
+    void register_differentUsers_Ok() {
+        User user1 = createUser(DEFAULT_AGE, DEFAULT_PASSWORD, DEFAULT_LOGIN);
+        User user2 = createUser(DEFAULT_AGE, DEFAULT_PASSWORD, DEFAULT_LOGIN + "a");
+        registrationService.register(user);
+        registrationService.register(user2);
+        User expectedUser1 = Storage.people.get(0);
+        User expectedUser2 = Storage.people.get(1);
```

Comment:
use `UserDao`

Reply:
I don't know what it is and how to use it / I think it's something with JDBC / but I can't find any normal information about it and how to use it

Reply:
<img width="1296" alt="image" src="https://user-images.githubusercontent.com/61326568/185899933-f2dd7a57-7f35-4d87-ac2a-2806ce9fb290.png">